### ISSUE TITLE
priority must be float not integer

### ISSUE THREAD

**Issue by liz-is** [daler/trackhub#20]
Hi,
I realised that if I assign an integer priority to a track the priority does not get rendered. I tracked this down to it failing validation because the priority parameter should be a float, not an int. Although it's easy enough for me to assign a float priority to my tracks, it's also easy to forget to do this. Would it be possible to coerce an integer priority to a float inside add_params? Or produce a warning when adding an invalid parameter, or something else to alert the user early on.
Thanks!

```
x.add_params(priority = 1)
print(x)
# track peaks_mysample_Rep1
# bigDataUrl peaks_mysample_Rep1.bigBed
# shortLabel mysample
# longLabel Peaks mysample_Rep1
# type bigBed
# color 222,45,38
# autoScale off

x.add_params(priority = 1.0)
print(x)
# track peaks_mysample_Rep1
# bigDataUrl peaks_mysample_Rep1.bigBed
# shortLabel mysample
# longLabel Peaks mysample_Rep1
# type bigBed
# color 222,45,38
# priority 1.0
# autoScale off
```

**Comment by daler** [comment]
Thanks for reporting this. I actually have validation built in (see `validate.py` and `constants.py`). For example:

```python
from trackhub import constants
constants.track_fields['priority'].validate(1)
# False

constants.track_fields['priority'].validate(1.0)
# True
```

But `BaseTrack.add_params` is just running the validation without paying attention to the True/False value returned. I forget if this is due to incomplete information on every parameter or just an oversight so I'll have to take a closer look.

**Comment by daler** [comment]
Closing as this was fixed in #48
